app/controllers/admin/base_controller.rb

```diff
@@ -1,13 +1,27 @@
 class Admin::BaseController < ApplicationController
   before_action :authenticate_user!
   before_action :check_admin
+  before_action :set_breadcrumbs
   layout "admin/admin_index"
 
   def index
     @courses = Course.all
     @users = User.all
   end
 
+  protected
```

Comment:
tại sao lại là protected?

Reply:
> tại sao lại là protected?

Không phải private vì để các controller khác kế thừa từ Admin::BaseController có thể sử dụng, Vì e muốn tách biệt hàm xử lý với các hàm public in ra view và để các hàm đó không được truy xuất bên ngoài controller khác mà không kế thừa nên e dùng protected